### PR TITLE
Oo/oslynn/15664/update rest transport for oracle rln etor nbs results

### PR DESCRIPTION
**Pull request by oslynn**
This PR update REST Transport to allow Oracle-rln.etor-nbs-results to get it's Api Key from the Vault.

Test Steps:
1. *Include steps to test these changes*

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/kotlin/transport/RESTTransport.kt [hunk 1]
```diff
@@ -111,8 +111,11 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
                         val httpHeaders = getHeaders(restTransportInfo, reportId)
                         var accessToken: String? = null
 
-                        if (restTransportInfo.authType == "apiKey") {
+                        if (restTransportInfo.authType == "Api Key") {
```
Comment (victor-chaparro) [inline prime-router/src/main/kotlin/transport/RESTTransport.kt:114]: We are using apiKey for ELIMS why do we need to change it to Api Key?

File: prime-router/src/main/kotlin/transport/RESTTransport.kt [hunk 2]
```diff
@@ -111,8 +111,11 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
...
                             val apiKeyCredential = credential as UserApiKeyCredential
+                            if (httpHeaders["shared-api-key"] == "From Vault") {
```
Comment (victor-chaparro) [inline prime-router/src/main/kotlin/transport/RESTTransport.kt:116]: Can we just create the httpHeader without checking if (httpHeaders["shared-api-key"] == "From Vault")?

**Review by victor-chaparro — CHANGES_REQUESTED** [review]

File: prime-router/src/main/kotlin/transport/RESTTransport.kt (hunk 1 above)
Comment (oslynn) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:114]: Using apiKey, Gitleaks doesn't allow me to commit.  I wonder, how did you commit with using "apiKey"?

File: prime-router/src/main/kotlin/transport/RESTTransport.kt (hunk 2 above)
Comment (oslynn) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:116]: Yes, we can do that.

File: prime-router/src/main/kotlin/transport/RESTTransport.kt (hunk 1 above)
Comment (basiliskus) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:114]: @oslynn You should be able to exclude/ignore that warning in the gitleaks config

File: prime-router/src/main/kotlin/transport/RESTTransport.kt (hunk 2 above)
Comment (victor-chaparro) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:116]: I didn't get any gitleak issues when commiting. Maybe you can disable it, I'll add it as a suggestion here and see if it lets you that way.

File: prime-router/src/main/kotlin/transport/RESTTransport.kt (hunk 1 above)
Comment (victor-chaparro) [inline prime-router/src/main/kotlin/transport/RESTTransport.kt:114]:
```suggestion
                        if (restTransportInfo.authType == "apiKey") {
```

File: prime-router/settings/STLTs/Oracle/oracle-rln.yml [hunk 3]
```diff
@@ -45,9 +45,21 @@
...
       transport:
-        type: "SFTP"
-        host: "172.17.6.20"
-        port: 22
-        filePath: "./upload"
-        credentialName: null
+        reportUrl: "https://spaces.erxhubdevcert.cerner.com/etor"
+        authTokenUrl: ""
+        encryptionKeyUrl: ""
+        authType: "Api Key"
```
Comment (victor-chaparro) [inline prime-router/settings/STLTs/Oracle/oracle-rln.yml:53]:
```suggestion
        authType: "apiKey"
```

File: prime-router/src/main/kotlin/transport/RESTTransport.kt (hunk 2 above)
Comment (oslynn) [reply prime-router/src/main/kotlin/transport/RESTTransport.kt:116]: Configure gitleaks for allow "apiKey",

**Review by victor-chaparro — APPROVED** [review]

**Review by snesm — APPROVED** [review]

**Pull request merged**